source/docs/software/commandbased/organizing-command-based.rst

```diff
@@ -0,0 +1,259 @@
+
+The easiest and most expressive way to do this is with a ``StartEndCommand``:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    Command runIntake = new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake);
+
+  .. code-tab:: c++
+
+    // TODO
+
+This is sufficient for commands that are only used once. However, for a command like this that might get used in many different autonomous routines and button bindings, inline commands everywhere means a lot of repetitive code:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // RobotContainer.java
+    intakeButton.whileHeld(new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake));
+
+
+    Command intakeAndShoot = new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake)
+        .alongWith(new RunShooter(shooter));
+
+    Command autonomousCommand = new ParallelCommandGroup(
+        new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake).withTimeout(5),
+        new WaitCommand(3),
+        new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake).withTimeout(5)
+    )
+
+  .. code-tab:: c++
+
+    // TODO
+
+(Creating one ``StartEndCommand`` instance and putting it in a variable won't work here, since once an instance of a command is added to a command group it is effectively "owned" by that command group and cannot be used in any other context.)
+
+Subclassing CommandBase
+^^^^^^^^^^^^^^^^^^^^^^^
+
+One possible solution to this problem is to create a new subclass of ``CommandBase`` that implements the necessary ``initialize`` and ``end`` routines.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    public class RunIntakeCommand extends CommandBase {
+        private Intake m_intake;
+
+        public RunIntakeCommand(Intake intake) {
+            this.m_intake = intake;
+            addRequirements(intake);
+        }
+
+        @Override
+        public void initialize() {
+            m_intake.setPercent(1);
+        }
+
+        @Override
+        public void end(boolean interrupted) {
+            m_intake.setPercent(0);
+        }
+
+        @Override
+        public boolean isFinished() {
+            return false;
+        }
+    }
+
+  .. code-tab:: c++
+
+    // TODO
+
+This, however, is just as cumbersome. The only two lines that really matter in this entire file are the two calls to ``intake.setPercent``, yet there are over 20 lines of boilerplate code! Not to mention, doing this for a lot of robot actions quickly clutters up a robot project with dozens of small files. Nevertheless, this might feel more "natural," particularly for programmers who don't feel confident with Java and want to stick closely to an object-oriented model.
+
+Factory Methods
+^^^^^^^^^^^^^^^
+
+A useful middle ground between these two extremes is using a `factory method <https://www.tutorialspoint.com/design_pattern/factory_pattern.htm>`__. A factory method is a method that, each time it is called, returns a new object according to some specification.
```

Comment:
I'm not sure about the link here; it seems to refer more to factory objects, which I'm not sure is fully relevant.